web-transport-quinn/src/client.rs

```diff
@@ -66,32 +90,48 @@ impl Client {
     }
 
     /// Connect to the server.
-    pub async fn connect(&self, url: &Url) -> Result<Session, ClientError> {
-        // Configure the crypto client.
-        let client_crypto = rustls::ClientConfig::builder();
-        let mut client_crypto = match &self.fingerprints {
-            Some(fingerprints) => client_crypto
-                .dangerous()
-                .with_custom_certificate_verifier(fingerprints.clone())
-                .with_no_client_auth(),
-            None => ClientConfig::with_platform_verifier(),
+    pub async fn connect(mut self, url: &Url) -> Result<Session, ClientError> {
```

Comment:
I don't think it should take `self`, or at the very least it should be `Clone` then. It's common to have one client config for multiple endpoints.